lib/spack/spack/solver/asp.py

```diff
@@ -878,6 +878,31 @@ def package_compiler_defaults(self, pkg):
                 fn.node_compiler_preference(pkg.name, cspec.name, cspec.version, -i * 100)
             )
 
+    def cfg_required_rules(self, pkg):
+        pkg_name = pkg.name
+        config = spack.config.get("packages")
+        requirements = config.get(pkg_name, {}).get("require", {})
+        if isinstance(requirements, string_types):
+            rules = [(pkg_name, [requirements])]
+        else:
+            rules = []
+            for requirement in requirements:
+                rules.append((pkg_name, requirement["one_of"]))
+        cnd_grp_id = 0
```

Comment:
(minor) Maybe this is slightly more readable:
```suggestion
        group_id = 0
```
?